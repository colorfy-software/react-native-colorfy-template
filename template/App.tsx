import React from 'react'
import { Provider } from 'react-redux'
import { View, StyleSheet } from 'react-native'
import { PersistGate } from 'redux-persist/integration/react'
import { ModalProvider, createModalStack } from 'react-native-modalfy'

import store, { persistor } from './src/redux/store'

declare let global: { HermesInternal: null | {} }

/**
 * @description Definition of the modal stack
 */
const modalStack = createModalStack({})

const App = (): JSX.Element => {
  return (
    <View style={styles.container}>
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <ModalProvider stack={modalStack}>
            <View />
          </ModalProvider>
        </PersistGate>
      </Provider>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'pink',
  },
})

export default App
