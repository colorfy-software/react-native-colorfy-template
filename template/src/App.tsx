import { useCallback, useEffect, useState } from 'react'
import { Animated, Keyboard, LayoutAnimation, Platform, UIManager, View } from 'react-native'
import { ModalOptions, ModalProvider, ModalStackConfig, createModalStack } from 'react-native-modalfy'

import core from './core/core'
import * as Modals from './modals'
import Navigator from './navigation/Navigation'

import { screen } from './styles/style-guide'

// NOTE: Definition of the modal stack
const modalConfig: ModalStackConfig = { ...Modals }

const animation = (animatedValue: Animated.Value, toValue: number) => {
  Animated.spring(animatedValue, {
    toValue,
    damping: 10,
    mass: 0.35,
    stiffness: 77.5,
    overshootClamping: true,
    restSpeedThreshold: 0.001,
    restDisplacementThreshold: 0.001,
    useNativeDriver: true,
  }).start()
}

const defaultOptions: ModalOptions = {
  animationIn: animation,
  animationOut: animation,
  position: 'bottom',
  transitionOptions: animatedValue => ({
    opacity: animatedValue.interpolate({
      inputRange: [0, 1, 2],
      outputRange: [0, 1, 0.9],
    }),
    transform: [
      {
        translateY: animatedValue.interpolate({
          inputRange: [0, 1, 2],
          outputRange: [screen.vh(100), 0, screen.vh(100)],
        }),
      },
    ],
  }),
}

const modalStack = createModalStack(modalConfig, defaultOptions)

const App = (): JSX.Element => {
  const [windowHeight, setWindowHeight] = useState(screen.screenHeight - screen.navBarHeight)

  // NOTE: Sending the available window height to all listeners on core.events.listen('windowHeight') channel.
  const syncWindowHeight = useCallback(height => {
    setWindowHeight(height - screen.navBarHeight)
    core.events.send('windowHeight', height - screen.navBarHeight)
  }, [])

  const keyboardWillShow = useCallback(
    (e: { endCoordinates: { height: number } }) => {
      LayoutAnimation.easeInEaseOut()
      syncWindowHeight(screen.screenHeight - e.endCoordinates.height)
    },
    [syncWindowHeight],
  )

  const keyboardWillHide = useCallback(() => {
    LayoutAnimation.easeInEaseOut()
    syncWindowHeight(screen.screenHeight)
  }, [syncWindowHeight])

  const keyboardDidShow = useCallback(
    (e: { endCoordinates: { height: number } }) => {
      Platform.OS === 'android' && syncWindowHeight(screen.screenHeight - e.endCoordinates.height)
    },
    [syncWindowHeight],
  )
  const keyboardDidHide = useCallback(() => {
    Platform.OS === 'android' && syncWindowHeight(screen.screenHeight)
  }, [syncWindowHeight])

  useEffect(() => {
    if (Platform.OS === 'android') {
      UIManager.setLayoutAnimationEnabledExperimental?.(true)
    }
  }, [])

  useEffect(() => {
    Keyboard.addListener('keyboardWillShow', keyboardWillShow)
    Keyboard.addListener('keyboardWillHide', keyboardWillHide)

    Keyboard.addListener('keyboardDidShow', keyboardDidShow)
    Keyboard.addListener('keyboardDidHide', keyboardDidHide)

    return () => {
      Keyboard.removeListener('keyboardWillShow', keyboardWillShow)
      Keyboard.removeListener('keyboardWillHide', keyboardWillHide)
      Keyboard.removeListener('keyboardDidShow', keyboardDidShow)
      Keyboard.removeListener('keyboardDidHide', keyboardDidHide)
    }
  }, [keyboardDidHide, keyboardDidShow, keyboardWillHide, keyboardWillShow])

  return (
    <View style={[{ width: screen.width(1), height: windowHeight }]}>
      <ModalProvider stack={modalStack}>
        <Navigator />
      </ModalProvider>
    </View>
  )
}

export default App
