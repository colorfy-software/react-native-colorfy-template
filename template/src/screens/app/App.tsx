import React from 'react'
import { View, StyleSheet } from 'react-native'

const App = (): JSX.Element => {
  return <View style={styles.container} />
}

export default App

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
})
