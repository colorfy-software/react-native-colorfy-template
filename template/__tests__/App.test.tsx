import { render } from '@testing-library/react-native'

import App from '../src/App'

describe('🌍 App:', () => {
  const { toJSON } = render(<App />)

  it('renders correctly', () => {
    expect(toJSON()).toMatchSnapshot()

    expect.assertions(1)
  })
})
