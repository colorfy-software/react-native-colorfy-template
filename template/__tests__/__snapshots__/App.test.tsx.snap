// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`üåç App: renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 1334,
        "width": 750,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              needsOffscreenAlphaCompositing={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <RNSScreenContainer
                          onLayout={[Function]}
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <RNSScreen
                            activityState={2}
                            forwardedRef={[Function]}
                            pointerEvents="box-none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            />
                            <View
                              accessibilityElementsHidden={false}
                              closing={false}
                              gestureVelocityImpact={0.3}
                              importantForAccessibility="auto"
                              onClose={[Function]}
                              onGestureBegin={[Function]}
                              onGestureCanceled={[Function]}
                              onGestureEnd={[Function]}
                              onOpen={[Function]}
                              onTransition={[Function]}
                              pointerEvents="box-none"
                              style={
                                Array [
                                  Object {
                                    "overflow": undefined,
                                  },
                                  Object {
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  },
                                ]
                              }
                              transitionSpec={
                                Object {
                                  "close": Object {
                                    "animation": "spring",
                                    "config": Object {
                                      "damping": 500,
                                      "mass": 3,
                                      "overshootClamping": true,
                                      "restDisplacementThreshold": 10,
                                      "restSpeedThreshold": 10,
                                      "stiffness": 1000,
                                    },
                                  },
                                  "open": Object {
                                    "animation": "spring",
                                    "config": Object {
                                      "damping": 500,
                                      "mass": 3,
                                      "overshootClamping": true,
                                      "restDisplacementThreshold": 10,
                                      "restSpeedThreshold": 10,
                                      "stiffness": 1000,
                                    },
                                  },
                                }
                              }
                            >
                              <View
                                needsOffscreenAlphaCompositing={false}
                                pointerEvents="box-none"
                                style={
                                  Object {
                                    "flex": 1,
                                  }
                                }
                              >
                                <View
                                  collapsable={false}
                                  forwardedRef={[Function]}
                                  onGestureHandlerEvent={[Function]}
                                  onGestureHandlerStateChange={[Function]}
                                  style={
                                    Object {
                                      "flex": 1,
                                      "transform": Array [
                                        Object {
                                          "translateX": 0,
                                        },
                                        Object {
                                          "translateX": 0,
                                        },
                                      ],
                                    }
                                  }
                                >
                                  <View
                                    pointerEvents="none"
                                    style={
                                      Object {
                                        "backgroundColor": "rgb(242, 242, 242)",
                                        "bottom": 0,
                                        "left": 0,
                                        "position": "absolute",
                                        "shadowColor": "#000",
                                        "shadowOffset": Object {
                                          "height": 1,
                                          "width": -1,
                                        },
                                        "shadowOpacity": 0.3,
                                        "shadowRadius": 5,
                                        "top": 0,
                                        "width": 3,
                                      }
                                    }
                                  />
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "overflow": "hidden",
                                        },
                                        Array [
                                          Object {
                                            "backgroundColor": "rgb(242, 242, 242)",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "flex": 1,
                                          "flexDirection": "column-reverse",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "flex": 1,
                                          }
                                        }
                                      >
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "height": 1334,
                                              "justifyContent": "space-around",
                                              "width": 750,
                                            }
                                          }
                                          testID="Login"
                                        >
                                          <View
                                            style={
                                              Object {
                                                "alignItems": "center",
                                                "justifyContent": "center",
                                              }
                                            }
                                          >
                                            <Image
                                              source={
                                                Object {
                                                  "testUri": "../../../src/assets/logo.png",
                                                }
                                              }
                                              style={
                                                Object {
                                                  "borderRadius": 100,
                                                  "height": 200,
                                                  "marginBottom": 49.285714285714285,
                                                  "width": 200,
                                                }
                                              }
                                              testID="Login.Logo"
                                            />
                                            <Text
                                              selectable={true}
                                              style={
                                                Array [
                                                  Object {
                                                    "fontFamily": "NotoSansSC-Bold",
                                                    "fontSize": 48,
                                                    "letterSpacing": -0.5,
                                                    "lineHeight": 29,
                                                  },
                                                  Object {
                                                    "fontSize": 48,
                                                  },
                                                  Object {},
                                                  Object {},
                                                  Object {},
                                                  Object {
                                                    "color": "#000000",
                                                  },
                                                ]
                                              }
                                            >
                                              unknown
                                            </Text>
                                          </View>
                                          <View
                                            accessible={true}
                                            collapsable={false}
                                            focusable={true}
                                            onBlur={[Function]}
                                            onClick={[Function]}
                                            onFocus={[Function]}
                                            onResponderGrant={[Function]}
                                            onResponderMove={[Function]}
                                            onResponderRelease={[Function]}
                                            onResponderTerminate={[Function]}
                                            onResponderTerminationRequest={[Function]}
                                            onStartShouldSetResponder={[Function]}
                                            style={
                                              Array [
                                                Object {
                                                  "alignItems": "center",
                                                  "borderWidth": 2,
                                                  "height": 60,
                                                  "justifyContent": "center",
                                                  "paddingHorizontal": 50,
                                                  "paddingVertical": 8.214285714285714,
                                                },
                                                Object {
                                                  "backgroundColor": "#e20e17",
                                                  "borderColor": "transparent",
                                                  "borderRadius": 8,
                                                  "opacity": 1,
                                                },
                                                undefined,
                                              ]
                                            }
                                            testID="Login.Button"
                                          >
                                            <Text
                                              numberOfLines={2}
                                              selectable={true}
                                              style={
                                                Array [
                                                  Object {
                                                    "fontFamily": "NotoSansSC-Medium",
                                                    "fontSize": 28,
                                                    "letterSpacing": -0.3,
                                                    "lineHeight": 16,
                                                  },
                                                  Object {
                                                    "fontSize": 28,
                                                  },
                                                  Array [
                                                    Object {
                                                      "fontSize": 32,
                                                      "lineHeight": 20,
                                                      "textAlign": "center",
                                                    },
                                                    undefined,
                                                  ],
                                                  Object {
                                                    "fontFamily": "NotoSansSC-Bold",
                                                  },
                                                  Object {},
                                                  Object {
                                                    "color": "white",
                                                  },
                                                ]
                                              }
                                            >
                                              Log in
                                            </Text>
                                          </View>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RNSScreen>
                        </RNSScreenContainer>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </RNCSafeAreaProvider>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "zIndex": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
